config/initializers/devise.rb

```diff
@@ -30,10 +30,55 @@
   warden.cookies.delete('_session_id')
 end
 
+module Devise
+  mattr_accessor :pam_authentication
+  @@pam_authentication = false
+  mattr_accessor :pam_service
+  @@pam_service = 'rpam'
+  module Strategies
+    class PamUserAuthenticatable < Authenticatable
+      def valid?
+        Devise.pam_authentication && params[scope] && params[scope][:email] && !params[scope][:email].index('@')
```

Comment:
People could have e-mails as LDAP identifiers, couldn't they?

I am against having both registrations and PAM *at the same time*, if PAM is active all logins should go through it - at least that's the LDAP principle from other PRs if I remember correctly.